### ISSUE TITLE
WorkerPool debug port limit reached: 65535>= 65535

### ISSUE THREAD

**Issue by JrmyAS** [josdejong/workerpool#318]
Hello everyone

I am using workerpool to generate vehicle trips with lots of steps from simpler route.
Everything works fine except that after a little more than 24 hours of processing on my server, this error occurs:
WorkerPool debug port limit reached: 65535> = 65535.

I am using workerpool with the proxy method.

Has this problem ever happened to you?

**Comment by josdejong** [comment]
I haven't heard about it before. How can I reproduce this issue?

**Comment by AlertesSpeciales** [comment]
I think I had done everything backwards.
I had placed `const pool = workerpool.pool('./core/systeme/worker_trajets.js');`
inside a function with each time a `pool.terminate()` at the end.

I put the const at the start of the code and I am now using pool.exec without pool.terminate at the end and have no more errors.

I think that at the end of the 65535 generation of trips, there was no longer any free debugport since I closed the previous one each time.

**Comment by josdejong** [comment]
Good to hear that you have a working solution now. 

I would expect that `pool.terminate()` would neatly cleanup and free any open ports again, so this should not happen I think (search the code for `DEBUG_PORT_ALLOCATOR.releasePort`). Anyone able to do some debugging here?

**Comment by panella87** [comment]
i'm having this issue every 4/5 hours

Error:
`
/node_modules/workerpool/src/debug-port-allocator.js:16
    throw new Error('WorkerPool debug port limit reached: ' + starting + '>= ' + MAX_P
ORTS );
          ^

Error: WorkerPool debug port limit reached: 65535>= 65535
    at DebugPortAllocator.nextAvailableStartingAt (/home/dev/volatility-seeker/node_mo
dules/workerpool/src/debug-port-allocator.js:16:11)
    at Pool._createWorkerHandler (/home/dev/volatility-seeker/node_modules/workerpool/
src/Pool.js:399:37)
    at Pool._getWorker (/home/dev/volatility-seeker/node_modules/workerpool/src/Pool.j
s:255:19)
    at Pool._next (/home/dev/volatility-seeker/node_modules/workerpool/src/Pool.js:202
:23)
    at Pool.exec (/home/dev/volatility-seeker/node_modules/workerpool/src/Pool.js:138:
10)
`

This is the code that calls the worker, pretty straightforward
`            
// creating worker pool
var pool = workerpool.pool(workerpoolExternalScriptPath);
// separate inversion negative to positive
array = await pool.exec("generateArraySimple", [filteredData])
// close worker pool
await pool.terminate(); // terminate all workers when done
`

i call the above lines around 2500 times every hour though:
i sent them in bulk of 10 and await for each one to complete so it should be 10 active pool at a time

How can i help the debugging process?

**Comment by panella87** [comment]
Hi there, i'm experiencing this behaviour too

I'm running my nodejs script on a server which have an extensive network workload that probably uses a lot of available ports

I'm shaping the code to create as less worker as possible but i still having this error once in a while:

```
/home/dev/v/node_modules/workerpool/src/debug-port-allocator.js:16
    throw new Error('WorkerPool debug port limit reached: ' + starting + '>= ' + MAX_PORTS );
          ^
Error: WorkerPool debug port limit reached: 65535>= 65535
    at DebugPortAllocator.nextAvailableStartingAt (/home/dev/v/node_modules/workerpool/src/debug-port-allocator.js:16:11)
    at Pool._createWorkerHandler (/home/dev/v/node_modules/workerpool/src/Pool.js:399:37)
```

It's pretty weird because i should be calling a maximum of 2 worker at a time, though after a few hours of activity i get the above error

Here's the code to create/destroy the worker

```
for (var i = 0; i < chunkedList.length; i++) {
// creating worker pool
var bulkScheduledActionWorkerPool = workerpool.pool(bulkScheduledActionWorkerScriptPath);  
// get data
var array = await bulkScheduledActionWorkerPool.exec("getMultipleInfoMongo", [mongoDBUrl, mongoDBName, mongoDBHistoryCollectionName, arrayOfHistoryDB_idInHexFormat, options.pipelineArgs])
// terminate all workers when done
await bulkScheduledActionWorkerPool.terminate();
}
```

Am i missing something? Available for debugging

Thanks in advance

**Comment by josdejong** [comment]
Thanks to your code snipped I found the cause: when terminating a workerpool, the debug ports where not released. This is fixed now in `v6.3.1` via 5d4309d9899fc99eb36eebb818aea1fa67fbe6e9.
